dbms/src/Flash/Disaggregated/RNPageReceiverContext.cpp

```diff
@@ -99,9 +102,36 @@ FetchPagesRequest::FetchPagesRequest(DM::RNRemoteSegmentReadTaskPtr seg_task_)
     *req->mutable_snapshot_id() = seg_task->snapshot_id.toMeta();
     req->set_table_id(seg_task->table_id);
     req->set_segment_id(seg_task->segment_id);
-    for (auto page_id : seg_task->cacheMissPageIds())
+
     {
-        req->add_page_ids(page_id);
+        std::vector<DM::Remote::PageOID> persisted_oids;
+        persisted_oids.reserve(seg_task->delta_tinycf_page_ids.size());
+        for (const auto & page_id : seg_task->delta_tinycf_page_ids)
+        {
+            auto page_oid = DM::Remote::PageOID{
+                .store_id = seg_task->store_id,
+                .table_id = seg_task->table_id,
+                .page_id = page_id,
+            };
+            persisted_oids.emplace_back(page_oid);
+        }
+
+        auto page_cache = seg_task->dm_context->db_context.getSharedContextDisagg()->rn_page_cache;
+        auto occupy_result = page_cache->occupySpace(persisted_oids, seg_task->delta_tinycf_page_sizes);
```

Comment:
```suggestion
        std::vector<DM::Remote::PageOID> to_read_oids;
        to_read_oids.reserve(seg_task->delta_tinycf_page_ids.size());
        for (const auto & page_id : seg_task->delta_tinycf_page_ids)
        {
            auto page_oid = DM::Remote::PageOID{
                .store_id = seg_task->store_id,
                .table_id = seg_task->table_id,
                .page_id = page_id,
            };
            to_read_oids.emplace_back(page_oid);
        }

        auto page_cache = seg_task->dm_context->db_context.getSharedContextDisagg()->rn_page_cache;
        auto occupy_result = page_cache->occupySpace(to_read_oids, seg_task->delta_tinycf_page_sizes);
```
name it as `to_read_oids`? `persisted_oids` make me think they have already been persisted in RN

Reply:
Yes, it is a naming mistake. Thanks for pointing out!